src/witan/models/dem/ccm/core/projection_loop.clj

```diff
@@ -11,37 +11,37 @@
    :columns (mapv #(s/one [(second %)] (format "col %s" (name (first %)))) col-vec)
    s/Keyword s/Any})
 
-(def HistoricPopnEstimates
-  (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:age s/Int]
-                           [:year s/Int] [:popn s/Int]]))
-
-(def StartingPopn
+(def PopnSchema
   (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:age s/Int]
                            [:year s/Int] [:popn s/Int]]))
 
 ;; Functions:
 (defworkflowfn ->starting-popn
   "Takes in a dataset of popn estimates and a year for the projection.
    Returns a dataset w/ the starting popn (popn for the previous year)."
-  {:witan/name :get-starting-popn
+  {:witan/name :ccm-core/get-starting-popn
    :witan/version "1.0"
-   :witan/input-schema {:historic-popn-estimates HistoricPopnEstimates}
-   :witan/param-schema {:year s/Int}
-   :witan/output-schema {:starting-popn StartingPopn}}
-  [{:keys [historic-popn-estimates]} {:keys [year]}]
-  {:starting-popn (i/query-dataset historic-popn-estimates {:year (dec year)})})
+   :witan/input-schema {:popn PopnSchema}
+   :witan/param-schema {:first-proj-year s/Int}
+   :witan/output-schema {:starting-popn PopnSchema}}
+  [{:keys [popn]} {:keys [first-proj-year]}]
+  (let [latest-yr (reduce max (i/$ :year popn))]
+    (if (>= latest-yr first-proj-year)
```

Comment:
Not sure we need the `ccm-core` function either but keeping it for now. Will remove it when I know it's useless.

Thanks, as I said during demo yesterday I wasn't sure it was the place for the exception throwing. I can definitely have this check outside of this function.